File: Python/quotes/model/svmService.py
import grpc

from dataclasses import asdict
from lib.proto.py.svm_expression_pb2_grpc import SVMServiceStub
from lib.proto.py.svm_expression_pb2 import (
    LinearSVCRequest,
    LinearSVRRequest,    
    SVCRequest
)

class SVMService(object):
    def __init__(self, url:str)->None:
        self.__channel = grpc.insecure_channel(url)
        self.__stub = SVMServiceStub(self.__channel)

    def linearSVCResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LinearSVCRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LinearSVCEvent(request_message)
            return asdict(response)
        
    def linearSVRResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = LinearSVRRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.LinearSVREvent(request_message)
            return asdict(response)
        
    def svcResponse(
            self, 
            data
        )->dict:
        with self.__channel:
            request_message = SVCRequest(
                x_drop_data = data.x_drop_data,
                y_drop_data = data.y_drop_data,
                size = data.size,
                random = data.random,
                key = data.key,
                sample_weight = data.sample_weight | None,
                kwargs = data.kwargs
            )
            response = self.__stub.SVCEvent(request_message)
            return asdict(response)

File: GoLang/quotes/model/svmService.go
package main

import (
	"context"
	"time"

	pb "github.com/RYANCOAL9999/AI_Go_Proto_File"
	"github.com/gin-gonic/gin"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)

type SVMGRPCClient struct {
	client pb.SVMServiceClient
}

func NewSVMGRPCClient(addr string) (*SVMGRPCClient, error) {
	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		return nil, err
	}
	return &SVMGRPCClient{
		client: pb.NewSVMServiceClient(conn),
	}, nil
}

func LinearSVCResponse(svmClient *SVMGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "linear svc"

		var data pb.LinearRequest
		if err := ReadRequestData(c, &data); err != nil {
			HandleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := svmClient.client.LinearSVCEvent(ctx, &data)
		if err != nil {
			HandleGRPCError(c, err, event_name)
			return
		}

		SendResponse(c, response, event_name)
	}
}

func LinearSVRResponse(svmClient *SVMGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "linear svr"

		var data pb.LinearRequest
		if err := ReadRequestData(c, &data); err != nil {
			HandleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := svmClient.client.LinearSVREvent(ctx, &data)
		if err != nil {
			HandleGRPCError(c, err, event_name)
			return
		}

		SendResponse(c, response, event_name)
	}
}

func SVCResponse(svmClient *SVMGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "svc"

		var data pb.SVCRequest
		if err := ReadRequestData(c, &data); err != nil {
			HandleBadRequestError(c, err, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := svmClient.client.SVCEvent(ctx, &data)
		if err != nil {
			HandleGRPCError(c, err, event_name)
			return
		}

		SendResponse(c, response, event_name)
	}
}
